File: app/src/main/java/de/bitb/astroskop/ui/main/circumstances/details/CircumstanceDetailsActivity.java
package de.bitb.astroskop.ui.main.circumstances.details;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;

import de.bitb.astroskop.ui.base.BaseActivity;

public class CircumstanceDetailsActivity extends BaseActivity {

    public static final String KEY_CIRCUMSTANCE_UUID = "key_circumstance_uuid";

    public static void startActivity(Context context, String uuid) {
        Intent intent = new Intent(context, CircumstanceDetailsActivity.class);
        intent.putExtra(KEY_CIRCUMSTANCE_UUID, uuid);
        context.startActivity(intent);
    }

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        showFragment(CircumstanceDetailsFragment.createInstance(getIntent().getStringExtra(KEY_CIRCUMSTANCE_UUID)), false);
    }
}

File: app/src/main/java/de/bitb/astroskop/ui/main/circumstances/details/CircumstanceDetailsActivity.kt
package de.bitb.astroskop.ui.main.circumstances.details

import android.content.Context
import android.content.Intent
import android.os.Bundle

import de.bitb.astroskop.ui.base.BaseActivity

class CircumstanceDetailsActivity : BaseActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        showFragment(CircumstanceDetailsFragment.createInstance(intent.getStringExtra(KEY_CIRCUMSTANCE_UUID)), false)
    }

    companion object {

        val KEY_CIRCUMSTANCE_UUID = "key_circumstance_uuid"

        fun startActivity(context: Context, uuid: String) {
            val intent = Intent(context, CircumstanceDetailsActivity::class.java)
            intent.putExtra(KEY_CIRCUMSTANCE_UUID, uuid)
            context.startActivity(intent)
        }
    }
}
